src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -104,110 +102,181 @@ def convert_iso8601_to_datetime(
     return parse(timestamp)
 
 
+DICT_SEP = "/"
+
+
+def _filter_nones(some_dict: dict) -> dict:
+    return {key: value for key, value in some_dict.items() if value is not None}
+
+
 class PartialSnapshot:
-    def __init__(self, snapshot: "Snapshot") -> None:
-        """Create a PartialSnapshot. If no snapshot is provided, the object is
-        a immutable POD, and any attempt at mutating it will raise an
-        UnsupportedOperationException."""
-        self._data: TPMap[str, Any] = pyrsistent.m()
-        self._snapshot = copy.copy(snapshot) if snapshot else None
+    def __init__(self, snapshot: Optional["Snapshot"] = None) -> None:
+        self._realization_states = defaultdict(dict)
+        """A shallow dictionary of realization states. The key is a string with
+        realization number, pointing to a dict with members active (bool),
+        start_time (datetime), end_time (datetime) and status (str)."""
 
-    def update_status(self, status: str) -> None:
-        self._apply_update(SnapshotDict(status=status))
+        self._step_states = defaultdict(dict)
```

Comment:
Out of context a bit comment, but do we need steps? As far as I understand we only have a single step.

Reply:
plan of refactor of this whole thing, so far planned for after fixes are merged - remove steps : )

Reply:
Very good! Then could you perhaps make an issue for step removal?

Reply:
here it is:

https://github.com/equinor/ert/issues/5866